presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -402,12 +403,17 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
 
             switch (node.getType()) {
                 case INNER:
+                    ImmutableList.Builder<Expression> predicates = ImmutableList.builder();
+                    predicates.add(joinPredicate);
+                    node.getCriteria().stream()
+                            .map(criteria -> new IsNotNullPredicate(criteria.getLeft().toSymbolReference()))
+                            .forEach(predicates::add);
                     InnerJoinPushDownResult innerJoinPushDownResult = processInnerJoin(
                             metadata,
                             inheritedPredicate,
                             leftEffectivePredicate,
                             rightEffectivePredicate,
-                            joinPredicate,
+                            combineConjuncts(metadata, predicates.build()),
```

Comment:
please also push `non null` predicates for outer joins.

Reply:
Have added for outer joins also